src/ExadelMentorship.BusinessLogic/CommandInvoker.cs

```diff
@@ -0,0 +1,70 @@
+﻿using ExadelMentorship.BusinessLogic.Exceptions;
+using ExadelMentorship.BusinessLogic.Features;
+using ExadelMentorship.BusinessLogic.Features.WeatherFeature;
+using ExadelMentorship.BusinessLogic.Interfaces;
+using ExadelMentorship.BusinessLogic.Models;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ExadelMentorship.BusinessLogic
+{
+    public class CommandInvoker
```

Comment:
`CommandInvoker`'s single responsibility (not SRP!) is to take generic type of `ICommand` and find appropriate `ICommandHandler` and execute `.Handle` method. So, you have to move out the other code from this class.

Can you please just make another separate PR just for CommandInvoker, `ICommand` and `ICommndHandler` interfaces? I'll approve right away. Then in another PRs you will bind your specific command implementation (thus smaller and self-sufficient PRs that will be also approved right away)